agent/src/workload/workload_control_loop.rs

```diff
@@ -326,17 +336,34 @@ impl WorkloadControlLoop {
         ErrorFunc: FnOnce(ControlLoopState<WorkloadId, StChecker>, WorkloadInstanceName, String) -> Fut
             + 'static,
     {
+        let host_file_path_mount_point_mappings =
+            match Self::handle_mount_point_creation(&control_loop_state).await {
+                Ok(mapping) => mapping,
+                Err(err) => {
+                    log::error!(
+                        "Failed to create workload files for workload '{}': '{}'",
+                        control_loop_state.instance_name(),
+                        err
+                    );
+                    return control_loop_state;
+                }
+            };
+
         let new_instance_name = control_loop_state.workload_spec.instance_name.clone();
 
         match control_loop_state
             .runtime
             .create_workload(
                 control_loop_state.workload_spec.clone(),
                 control_loop_state.workload_id.clone(),
-                control_loop_state.control_interface_path.clone(),
+                control_loop_state
+                    .control_interface_path
+                    .clone()
+                    .map(|path| path.to_path_buf()),
```

Comment:
From the documentation of `to_path_buf`: "Converts a Path to an owned PathBuf.
To avoid the double cloning we should take a reference to the option:
```suggestion
                    .as_ref()
                    .map(|path| path.to_path_buf()),
```